src/quest/historian.py

```diff
@@ -626,7 +621,12 @@ async def record_external_event(self, name, identity, action, *args, **kwargs):
 
         logging.debug(f'External event {step_id} with {args} and {kwargs}')
 
-        resource = self._resources[resource_id]['resource']
+        try:
+            resource = self._resources[resource_id]['resource']
+        except KeyError:
+            resource_id = _create_resource_id(name, None)  # Try accessing by public identity
+            resource = self._resources[resource_id]['resource']
+
```

Comment:
I think for posting to a resource, it is ok for the caller to specify the exact identity that owns the resource (including public). 

But when posting to a resource stream, we should publish events for all resources a caller has access to.